### PR TITLE
Fix memory leak, update bstats version

### PR DESCRIPTION
**Pull request by sgdc3**

### CODE REVIEW

**Comment by kernitus** [comment]
Could you please make this PR only about the memory leak, and also provide a brief description about what you did there?

File: src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java [hunk 1]
```diff
@@ -55,6 +56,12 @@ public void onPlayerChangeWorld(PlayerChangedWorldEvent e){
         createOrUpdateTeam(e.getPlayer());
     }
 
+    @EventHandler
+    public void onPlayerQuit(PlayerQuitEvent e) {
+        PacketManager.getInstance().removeListener(collisionPacketListener, e.getPlayer());
+        playerTeamMap.remove(e.getPlayer());
```
Comment (I-Al-Istannen) [inline src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java:62]: Not strictly needed, as that is a [WeakHashMap](https://github.com/kernitus/BukkitOldCombatMechanics/blob/59ecdc25d9d6e59a20d9872f4f1a7a82fade40c9/src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java#L32).

File: src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java [hunk 2]
```diff
@@ -55,6 +56,12 @@ public void onPlayerChangeWorld(PlayerChangedWorldEvent e){
         createOrUpdateTeam(e.getPlayer());
     }
 
+    @EventHandler
+    public void onPlayerQuit(PlayerQuitEvent e) {
+        PacketManager.getInstance().removeListener(collisionPacketListener, e.getPlayer());
```
Comment (I-Al-Istannen) [inline src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java:61]: I don't think this is needed, [as the PacketManager should handle that for you.](https://github.com/kernitus/BukkitOldCombatMechanics/blob/59ecdc25d9d6e59a20d9872f4f1a7a82fade40c9/src/main/java/kernitus/plugin/OldCombatMechanics/utilities/packet/PacketManager.java#L144)

File: pom.xml [hunk 3]
```diff
@@ -30,6 +30,10 @@
             <id>local-file-repo</id>
             <url>file://${basedir}/internal-file-repo</url>
         </repository>
+        <repository>
+            <id>CodeMC</id>
```
Comment (I-Al-Istannen) [inline pom.xml:34]: Why was this added and what does it pull in? Adding maven repos and dependencies is a bit risky, so a small comment would be greatly appreciated :)

File: src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java (hunk 1 above)
Comment (sgdc3) [reply src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java:62]: It is always best practice to clean collections properly, you are still keeping references to TeamPacket objects.

File: src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java (hunk 2 above)
Comment (sgdc3) [reply src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java:61]: yeah, i just wasn't sure, it is a bit safer as packet listeners might be invoked async

File: src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java (hunk 1 above)
Comment (I-Al-Istannen) [reply src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java:62]: > Hash table based implementation of the Map interface, with weak keys. An entry in a WeakHashMap will automatically be removed when its key is no longer in ordinary use. More precisely, the presence of a mapping for a given key will not prevent the key from being discarded by the garbage collector, that is, made finalizable, finalized, and then reclaimed. **When a key has been discarded its entry is effectively removed from the map**, so this class behaves somewhat differently from other Map implementations. 

Where does the reference to the TeamPacket object survive?

File: pom.xml (hunk 3 above)
Comment (sgdc3) [reply pom.xml:34]: Bstats deprecated the old maven repository https://bstats.org/getting-started/include-metrics

Comment (I-Al-Istannen) [reply pom.xml:34]: Sounds good :)

**Comment by sgdc3** [comment]
I ha to upgrade bstats as the old 1.2 version is no longer available on the bstats repository.

File: src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java (hunk 2 above)
Comment (I-Al-Istannen) [reply src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java:61]: It might make it a bit more explicit, but the other event listens for PlayerQuit too on MONITOR priority.

`removeListener` should be fine with being called twice, I don't know if it is nicer to explicitly do that here again though.

File: src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java (hunk 1 above)
Comment (sgdc3) [reply src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java:62]: hmm, fair

Comment (I-Al-Istannen) [reply src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java:62]: It is probably still nice to have it though. As we just saw that code might be a bit too clever for its own good.

**Comment by I-Al-Istannen** [comment]
Sounds nice, making some things more explicit is probably helpful here :)

File: src/main/java/kernitus/plugin/OldCombatMechanics/OCMMain.java [hunk 4]
```diff
@@ -3,6 +3,7 @@
 import kernitus.plugin.OldCombatMechanics.hooks.PlaceholderAPIHook;
 import kernitus.plugin.OldCombatMechanics.hooks.api.Hook;
 import kernitus.plugin.OldCombatMechanics.module.*;
+import kernitus.plugin.OldCombatMechanics.module.Module;
```
Comment (Rayzr522) [inline src/main/java/kernitus/plugin/OldCombatMechanics/OCMMain.java:6]: why is this import required? we are importing `.module.*` on the line above already :thinking:

Comment (Rayzr522) [reply src/main/java/kernitus/plugin/OldCombatMechanics/OCMMain.java:6]: this should be removed

File: src/main/java/kernitus/plugin/OldCombatMechanics/module/ModuleSwordBlocking.java [hunk 5]
```diff
@@ -105,6 +105,7 @@ public void onWorldChange(PlayerChangedWorldEvent e){
     @EventHandler(priority = EventPriority.HIGHEST)
     public void onPlayerLogout(PlayerQuitEvent e){
         restore(e.getPlayer());
+        tryCancelTask(e.getPlayer().getUniqueId());
```
Comment (Rayzr522) [inline src/main/java/kernitus/plugin/OldCombatMechanics/module/ModuleSwordBlocking.java:108]: good catch, thanks :)

File: src/main/java/kernitus/plugin/OldCombatMechanics/OCMMain.java (hunk 4 above)
Comment (sgdc3) [reply src/main/java/kernitus/plugin/OldCombatMechanics/OCMMain.java:6]: It was conflicting with java.util.Module, i had to compile it under jdk 11 so my ide was complaining

**Comment by sgdc3** [comment]
The leak isn't fixed... something must be wrong in the SwordBlocking logic

**Comment by Rayzr522** [comment]
@sgdc3 where is the source of the memory leak? from somewhere in these maps?
```java
    private final Map<UUID, ItemStack> storedOffhandItems = new HashMap<>();
    private final Map<UUID, RunnableSeries> correspondingTasks = new HashMap<>();
```

and also, how are you detecting this memory leak? a heapdump? if so, a copy of that (if it's a reasonable size) or just a screenshot of it at least would be helpful :)

**Comment by sgdc3** [comment]
OCM is holding direct player references in 
Cattura

**Comment by sgdc3** [comment]
@Rayzr522 do you need any other data from the heap dump?
